go/integration_test/processor/processor_gas_test.go

```diff
@@ -145,132 +143,225 @@ func getGasTestScenarios() map[string]Scenario {
-			GasUsed: floria.TxGas + executionGasCost + 100/10,
-		},
-	}
-	allTestCases["SimpleCodeInsufficient"] = Scenario{
-		Before: WorldState{
-			{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
-			{2}: Account{Balance: tosca.NewValue(0),
-				Code: tosca.Code{
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.RETURN),
+		"SimpleCodeInsufficient": {
+			Before: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
 				},
 			},
-		},
-		Transaction: tosca.Transaction{
-			Sender:    tosca.Address{1},
-			Recipient: &tosca.Address{2},
-			GasLimit:  floria.TxGas + executionGasCost - 1,
-			Nonce:     4,
-		},
-		After: WorldState{
-			{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
-			{2}: Account{Balance: tosca.NewValue(0),
-				Code: tosca.Code{
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.PUSH1), byte(0), // < PUSH 0
-					byte(op.RETURN),
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{1},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + executionGasCost - 1,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{1}: Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0),
+					Code: tosca.Code{
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.PUSH1), byte(0), // < PUSH 0
+						byte(op.RETURN),
+					},
 				},
 			},
+			Receipt: tosca.Receipt{
+				Success: false,
+				GasUsed: floria.TxGas + executionGasCost - 1,
+			},
+			OperaError: fmt.Errorf("gas too low"),
 		},
-		Receipt: tosca.Receipt{
-			Success: false,
-			GasUsed: floria.TxGas + executionGasCost - 1,
+		"InternalCallDoesNotConsume10PercentOfRemainingGas": {
+			Before: WorldState{
+				{}:  Account{Balance: tosca.NewValue(100), Nonce: 4},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Transaction: tosca.Transaction{
+				Sender:    tosca.Address{},
+				Recipient: &tosca.Address{2},
+				GasLimit:  floria.TxGas + 100,
+				Nonce:     4,
+			},
+			After: WorldState{
+				{}:  Account{Balance: tosca.NewValue(100), Nonce: 5},
+				{2}: Account{Balance: tosca.NewValue(0)},
+			},
+			Receipt: tosca.Receipt{
+				Success: true,
+				GasUsed: floria.TxGas,
+			},
 		},
-		OperaError: fmt.Errorf("gas too low"),
 	}
 
-	allTestCases["InternalCallDoesNotConsume10RemainingPercentGas"] = Scenario{
-		Before: WorldState{
-			{}:  Account{Balance: tosca.NewValue(100), Nonce: 4},
-			{2}: Account{Balance: tosca.NewValue(0)},
-		},
-		Transaction: tosca.Transaction{
-			Sender:    tosca.Address{},
-			Recipient: &tosca.Address{2},
-			GasLimit:  floria.TxGas + 100,
-			Nonce:     4,
+	return testCases
+}
+
+func gasPricingTestCases() map[string]Scenario {
+	gasPrice := uint64(10)
+	testCases := map[string]Scenario{
```

Comment:
I would try to minimize the details that need to be specified in those cases. Everything that is the same in all of those should be handled in the Run function below.

Here, it looks like all you need is the before and after value for the account's balance and the receipt.

Reply:
I was able to reduce the unit test length and also restructured parts of the other gas specific tests.